Boards.h

```diff
@@ -666,24 +666,23 @@ writePort(port, value, bitmask):  Write an 8 bit port.
 #define PIN_TO_SERVO(p)         ((p) - 2)
 
 // ESP8266
-// hardware: TX0=1, TX1=2, RX0=3, MISO=12, MOSI=13, SCLK=14, CS=15, A0=17  
-// board depended defaults: LED_BUILTIN, SDA, SCL
-// note: boot mode GPIOs 0, 2 and 15 can be used as outputs (GPIOs 6-11 are in use for flash IO)
+// note: boot mode GPIOs 0, 2 and 15 can be used as outputs, GPIOs 6-11 are in use for flash IO
 #elif defined(ESP8266)
-#define TOTAL_ANALOG_PINS       1
-#define TOTAL_PINS              18 // 11 digital + 1 analog + 6 inaccessible
+#define TOTAL_ANALOG_PINS       NUM_ANALOG_INPUTS
+#define TOTAL_PINS              (NUM_DIGITAL_PINS > A0 + NUM_ANALOG_INPUTS ? NUM_DIGITAL_PINS : A0 + NUM_ANALOG_INPUTS)
+#define VERSION_BLINK_PIN       LED_BUILTIN
 #define PIN_SERIAL_RX           3
 #define PIN_SERIAL_TX           1
-#define IS_PIN_DIGITAL(p)       (((p) >= 0 && (p) <= 5) || ((p) >= 12 && (p) <= 16))
-#define IS_PIN_ANALOG(p)        ((p) == A0)
+#define IS_PIN_DIGITAL(p)       (((p) >= 0 && (p) <= 5) || ((p) >= 12 && (p) < (NUM_DIGITAL_PINS > A0 + NUM_ANALOG_INPUTS ? NUM_DIGITAL_PINS : A0)))
```

Comment:
should be able to change back to: `#define IS_PIN_DIGITAL(p)       ((p) >= 0 && (p) < A0)`